src/Pages/Recommend/RecommendComponent/RecommendModalTap.js

```diff
@@ -0,0 +1,169 @@
+import RecommendModalSecond from "../RecommendComponent/RecommendModalTapComponent/RecommendModalSecond";
+import RecommendModalThird from "../RecommendComponent/RecommendModalTapComponent/RecommendModalThird";
+import RecommendModalFourth from "../RecommendComponent/RecommendModalTapComponent/RecommendModalFourth";
+import "../RecommendComponent/Recommend.scss";
+import "../RecommendComponent/RecommendModalTap.scss";
+import styled from "styled-components";
+
+const tapObj = {
+  0: <RecommendModalFirst />,
+  1: <RecommendModalSecond />,
+  2: <RecommendModalThird />,
+  3: <RecommendModalFourth />
+};
+
+class RecommendModalTap extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      activeTap: 0
+    };
+  }
+
+  clickModalTapHandler = id => {
+    this.setState({
+      activeTap: id
+    });
+  };
+
+  render() {
+    return (
+      <>
+        <ModalTapBox>
+          <div className="modalTapOveray"></div>
+          <div className="modalTapBoard">
+            <header>
+              <span>추천 요청</span>
+              <button onClick={this.props.visibleState}>X</button>
+            </header>
+            <div className="modalTapContent">
+              <div className="tapTitle">
+                <button
+                  onClick={() => this.clickModalTapHandler(0)}
+                  className={
+                    this.state.activeTap === 0
+                      ? "clickModalBtn"
+                      : "unClickModalBtn"
+                  }
+                >
+                  링크
+                </button>
+                <button
+                  onClick={() => this.clickModalTapHandler(1)}
+                  className={
+                    this.state.activeTap === 1
+                      ? "clickModalBtn"
+                      : "unClickModalBtn"
+                  }
+                >
+                  이메일
+                </button>
+                <button
+                  onClick={() => this.clickModalTapHandler(2)}
+                  className={
+                    this.state.activeTap === 2
+                      ? "clickModalBtn"
+                      : "unClickModalBtn"
+                  }
+                >
+                  SMS
+                </button>
+                <button
+                  onClick={() => this.clickModalTapHandler(3)}
+                  className={
+                    this.state.activeTap === 3
+                      ? "clickModalBtn"
+                      : "unClickModalBtn"
+                  }
+                >
+                  원티드 친구
+                </button>
+              </div>
+              {(this.state.activeTap === 0 && <RecommendModalFirst />) ||
+                (this.state.activeTap === 1 && <RecommendModalSecond />) ||
+                (this.state.activeTap === 2 && <RecommendModalThird />) ||
+                (this.state.activeTap === 3 && <RecommendModalFourth />)}
```

Comment:
activeTap의 값이 겹칠 일이 없어서 ```||``` 이 논리 연산자는 불필요한 것 같습니다
그리고 위에서 정의한 변수를 바로 사용할 수 있습니다!
```suggestion
   {tapObj[this.state.activeTap]}
```

Reply:
수정 완료 했습니다.